### PR TITLE
Don't require passwords local networks

### PR DESCRIPTION
**Pull request by brianjmurrell**
Proposed changes in this pull request:
- when being accessed from a host on the local network, don't require
the username/password that was instituted to protect access from the
Internet

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickChill/SickChill/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: sickbeard/helpers.py [hunk 1]
```diff
@@ -1745,6 +1746,27 @@ def is_ip_private(ip):
...
+def is_ip_local(ip):
+    for adapter in ifaddr.get_adapters():
+        for aip in adapter.ips:
+            #print("   %s/%s" % (aip.ip, aip.network_prefix))
+            network = None
+            if isinstance(aip.ip, tuple):
+                network = ipaddress.IPv6Network("%s/%s" % (aip.ip[0],
+                                                           aip.network_prefix),
+                                                strict=False)
+                if ipaddress.ip_address(addr) in network:
```
Comment (miigotu) [inline sickbeard/helpers.py:1758]: addr is undefined

File: sickbeard/helpers.py [hunk 2]
```diff
@@ -1745,6 +1746,27 @@ def is_ip_private(ip):
...
+                    return True
+
+            else:
+                network = ipaddress.IPv4Network("%s/%s" % (aip.ip,
+                                                           aip.network_prefix),
+                                                strict=False)
+                if ipaddress.ip_address(addr) in network:
```
Comment (miigotu) [inline sickbeard/helpers.py:1765]: addr is undefined

**Review by miigotu — CHANGES_REQUESTED** [review]

File: sickbeard/helpers.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 import hashlib
 import io
 import ipaddress
+import ifaddr
```
Comment (miigotu) [inline sickbeard/helpers.py:32]: image

**Review by miigotu — CHANGES_REQUESTED** [review]
Also, add ifaddr to requirements.txt

**Comment by miigotu** [comment]
I made the changes for you, you can see what changes I made here: https://github.com/SickChill/SickChill/compare/de0c3c30987ff39a5788ccb686e8e7a45ed508b9..55c9910e03d1d19921306348257b2f909049b6b6

Thanks for the PR, good find on that lib it solves quite a bit of problems.

**Comment by brianjmurrell** [comment]
> I made the changes for you, you can see what changes I made here: https://github.com/SickChill/SickChill/compare/de0c3c30987ff39a5788ccb686e8e7a45ed508b9..55c9910e03d1d19921306348257b2f909049b6b6

Looks good.  Thanks for the fixups.
 
> Thanks for the PR, good find on that lib it solves quite a bit of problems.

Much welcome.

**Comment by miigotu** [comment]
I might have to add back is_ip_private x.X it seems this misses quite a few cases such as docker and VPN??

**Pull request merged**